tests/test_needle_plot.py

```diff
@@ -0,0 +1,223 @@
+import os
+from tests.dashbio_demos.app_needle_plot import load_mutation_data, DATAPATH, DEMO_DATA
+from .test_common_features import init_demo_app, template_test_component_single_prop, \
+    PROP_TYPES, COMPONENT_REACT_BASE, generate_subprop_test
+
+APP_NAME = os.path.basename(__file__).replace('test_', '').replace('.py', '').replace('_', '-')
+
+TEST_DATA = load_mutation_data('{}{}'.format(DATAPATH, DEMO_DATA[0]['mutData']))
+
+
+# Demo app tests
+@init_demo_app(APP_NAME)
+def test_click_app_link_from_gallery(dash_threaded, selenium):
+
+    assert selenium.current_url.replace('http://localhost:8050', '').strip('/') == \
+           'dash-bio/{}'.format(APP_NAME)
+
+
+# Needle Plot component tests
+def needle_plot_test_props_callback(
+        nclicks,
+        p_name,
+        p_value,
+        prop_type=None,
+):
+    """Callback on a single user chosen prop on NeedlePlot component.
+        :param nclicks: (int) html.Button 'n_clicks' Input
+        :param p_name: (string) stands for prop name, dcc.Input 'value' State (not used here)
+        :param p_value: (string) stands for prop value, dcc.Input 'value' State
+        :param prop_type: (string) one of PARAM_TYPES keys
+            default: None
+        :return: the value of the prop to the dash.dependencies.Ouput()
+    """
+    answer = None
+    if prop_type == 'dict':
+        answer = {}
+    # avoid triggering at the creation of the button in the layout
+    if nclicks is not None:
+        # convert the parameter value to the right type
+        if prop_type in PROP_TYPES:
+            p_value = PROP_TYPES[prop_type](p_value)
+        answer = p_value
+    return answer
+
+
+def generate_subprop_test_needle(
+        dash_threaded,
+        selenium,
+        prop,
+        subprop,
+        subprop_type,
+        subprop_val
+):
+    """Avoid repetition of arguments which will be the same for a series of tests."""
```

Comment:
This is great information, but I don't think it describes *what* the function is actually doing (it's describing *why*). So I guess it would be along the lines of "Return a list of ... to avoid ..."